commit 089f245c75ec7269220bc13347b1264b54c0a16d
Author: JiapengLi <gapleehit@gmail.com>
Date:   Tue Apr 1 22:37:10 2014 +0800

    [ramips][mt7620a]Add support for Baidu M-100(A router with audio output)

diff --git a/target/linux/ramips/base-files/etc/diag.sh b/target/linux/ramips/base-files/etc/diag.sh
index 4168d0a..198a057 100755
--- a/target/linux/ramips/base-files/etc/diag.sh
+++ b/target/linux/ramips/base-files/etc/diag.sh
@@ -58,6 +58,9 @@ get_status_led() {
 	hw550-3g)
 		status_led="hw550-3g:green:status"
 		;;
+	m-100)
+		status_led="baidu:red:status"
+		;;
 	m3)
 		status_led="m3:blue:status"
 		;;
diff --git a/target/linux/ramips/base-files/etc/uci-defaults/01_leds b/target/linux/ramips/base-files/etc/uci-defaults/01_leds
index 567835e..a7a0b44 100755
--- a/target/linux/ramips/base-files/etc/uci-defaults/01_leds
+++ b/target/linux/ramips/base-files/etc/uci-defaults/01_leds
@@ -103,6 +103,9 @@ case $board in
 	hw550-3g)
 		set_usb_led "hw550-3g:green:usb"
 		;;
+	m-100)
+		set_wifi_led "baidu:blue:wlan"
+		;;
 	mofi3500-3gn)
 		set_usb_led "mofi3500-3gn:green:usb"
 		;;
diff --git a/target/linux/ramips/base-files/lib/ramips.sh b/target/linux/ramips/base-files/lib/ramips.sh
index a3a26cb..57ef110 100755
--- a/target/linux/ramips/base-files/lib/ramips.sh
+++ b/target/linux/ramips/base-files/lib/ramips.sh
@@ -64,6 +64,9 @@ ramips_board_detect() {
 	*"Aztech HW550-3G")
 		name="hw550-3g"
 		;;
+	*"BAIDU M-100")
+		name="m-100"
+		;;
 	*"DCS-930")
 		name="dcs-930"
 		;;
diff --git a/target/linux/ramips/base-files/lib/upgrade/platform.sh b/target/linux/ramips/base-files/lib/upgrade/platform.sh
index 8ff6063..f8c8672 100755
--- a/target/linux/ramips/base-files/lib/upgrade/platform.sh
+++ b/target/linux/ramips/base-files/lib/upgrade/platform.sh
@@ -50,6 +50,7 @@ platform_check_image() {
 	hg255d | \
 	hlk-rm04 | \
 	ip2202 | \
+	m-100 | \
 	m3 | \
 	m4 | \
 	mofi3500-3gn | \
diff --git a/target/linux/ramips/dts/M-100.dts b/target/linux/ramips/dts/M-100.dts
new file mode 100644
index 0000000..74cd7ef
--- /dev/null
+++ b/target/linux/ramips/dts/M-100.dts
@@ -0,0 +1,129 @@
+/dts-v1/;
+
+/include/ "mt7620a.dtsi"
+
+/ {
+	compatible = "M-100", "ralink,mt7620a-soc";
+	model = "BAIDU M-100";
+
+	palmbus@10000000 {
+		gpio0: gpio@600 {
+			status = "okay";
+		};
+		gpio1: gpio@638 {
+			status = "okay";
+		};
+		gpio2: gpio@660 {
+			status = "okay";
+		};
+		gpio3: gpio@688 {
+			status = "okay";
+		};
+
+		i2s@a00 {
+			status = "okay";
+		};
+
+		spi@b00 {
+			status = "okay";
+
+			m25p80@0 {
+				#address-cells = <1>;
+				#size-cells = <1>;
+				compatible = "en25q64";
+				reg = <0 0>;
+				linux,modalias = "m25p80", "en25q64";
+				spi-max-frequency = <10000000>;
+
+				partition@0 {
+					label = "u-boot";
+					reg = <0x0 0x30000>;
+					read-only;
+				};
+
+				partition@30000 {
+					label = "u-boot-env";
+					reg = <0x30000 0x10000>;
+					read-only;
+				};
+
+				factory: partition@40000 {
+					label = "factory";
+					reg = <0x40000 0x10000>;
+					read-only;
+				};
+
+				partition@50000 {
+					label = "firmware";
+					reg = <0x50000 0x7b0000>; // 8M-0x7b0000; 16M-0xFb0000;
+				};
+			};
+		};
+
+		gdma@2800 {
+			status = "okay";
+		};
+
+	};
+
+	pinctrl {
+		state_default: pinctrl0 {
+			gpio {
+				ralink,group = "wled", "mdio", "i2c", "rgmii1", "rgmii2";
+				ralink,function = "gpio";
+			};
+			i2s_gpio {
+				ralink,group = "uartf";
+				ralink,function = "gpio i2s";
+			};
+		};
+	};
+
+	ethernet@10100000 {
+		status = "okay";
+		mtd-mac-address = <&factory 0x4>;
+		ralink,port-map = "llllw";
+	};
+
+	gsw@10110000 {
+		ralink,port4 = "gmac";
+	};
+
+	sdhci@10130000 {
+		status = "okay";
+	};
+
+	pcie@10140000 {
+		status = "okay";
+	};
+
+	gpio-leds {
+		compatible = "gpio-leds";
+		status {
+			label = "baidu:red:status";
+			gpios = <&gpio0 11 0>;
+		};
+		wlan {
+			label = "baidu:blue:wlan";
+			gpios = <&gpio3 0 1>;
+		};
+	};
+
+	gpio-keys-polled {
+		compatible = "gpio-keys";
+		#address-cells = <1>;
+		#size-cells = <0>;
+		poll-interval = <20>;
+		reset {
+			label = "reset";
+			gpios = <&gpio0 13 1>;
+			linux,code = <0x198>;
+		};
+		wps {
+			label = "wps";
+			gpios = <&gpio1 3 1>;	//GPIO27
+			linux,code = <0x211>;
+		};
+	};
+
+};
diff --git a/target/linux/ramips/dts/mt7620a.dtsi b/target/linux/ramips/dts/mt7620a.dtsi
index b990ed3..2ade69a 100644
--- a/target/linux/ramips/dts/mt7620a.dtsi
+++ b/target/linux/ramips/dts/mt7620a.dtsi
@@ -282,6 +282,12 @@
 		pinctrl-0 = <&state_default>;
 		state_default: pinctrl0 {
 		};
+		i2s_gpio_pins: i2s_gpio {
+			i2s_gpio {
+				ralink,group = "uartf";
+				ralink,function = "gpio i2s";
+			};
+		};
 		pcm_i2s_pins: pcm_i2s {
 			pcm_i2s {
 				ralink,group = "uartf";
diff --git a/target/linux/ramips/image/Makefile b/target/linux/ramips/image/Makefile
index c6a42ad..22afee1 100644
--- a/target/linux/ramips/image/Makefile
+++ b/target/linux/ramips/image/Makefile
@@ -661,6 +661,7 @@ Image/Build/Profile/MT7620a_MT7610e=$(call BuildFirmware/Default8M/$(1),$(1),mt7
 Image/Build/Profile/MT7620a_MT7530=$(call BuildFirmware/Default8M/$(1),$(1),mt7620a_mt7530,MT7620a_MT7530)
 Image/Build/Profile/MT7620a_V22SG=$(call BuildFirmware/Default8M/$(1),$(1),mt7620a_v22sg,MT7620a_V22SG)
 Image/Build/Profile/RP-N53=$(call BuildFirmware/Default8M/$(1),$(1),rp_n53,RP-N53)
+Image/Build/Profile/M-100=$(call BuildFirmware/Default8M/$(1),$(1),m-100,M-100)
 whr_300hp2_mtd_size=7012352
 Image/Build/Profile/WHR300HP2=$(call BuildFirmware/CustomFlash/$(1),$(1),whr-300hp2,WHR-300HP2,$(whr_300hp2_mtd_size))
 Image/Build/Profile/WHR600D=$(call BuildFirmware/CustomFlash/$(1),$(1),whr-600d,WHR-600D,$(whr_300hp2_mtd_size))
@@ -680,6 +681,7 @@ define Image/Build/Profile/Default
 	$(call Image/Build/Profile/WHR300HP2,$(1))
 	$(call Image/Build/Profile/WHR600D,$(1))
 	$(call Image/Build/Profile/NA930,$(1))
+	$(call Image/Build/Profile/M-100,$(1))
 endef
 endif
 
diff --git a/target/linux/ramips/mt7620a/profiles/baidu.mk b/target/linux/ramips/mt7620a/profiles/baidu.mk
new file mode 100644
index 0000000..7029517
--- /dev/null
+++ b/target/linux/ramips/mt7620a/profiles/baidu.mk
@@ -0,0 +1,23 @@
+#
+# Copyright (C) 2013 OpenWrt.org
+#
+# This is free software, licensed under the GNU General Public License v2.
+# See /LICENSE for more information.
+# kmod-i2c-ralink \
+
+define Profile/M-100
+	NAME:=BAIDU M-100
+	PACKAGES:=\
+		kmod-usb-core kmod-usb-dwc2 \
+		kmod-ledtrig-usbdev \
+		kmod-sound-core kmod-sound-mt7620 \
+		kmod-input-core madplay mdnsresponder libspeex mpd mpc \
+		alsa-lib alsa-utils alsa-utils-tests libffmpeg-mini minidlna\
+		luci luci-app-minidlna 
+endef
+
+define Profile/M-100/Description
+	Package set for Baidu M-100 Product
+endef
+
+$(eval $(call Profile,M-100))
